packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -105,125 +103,120 @@ export class BlockSwitcher extends Component {
 			);
 		}
 
-		return (
-			<Dropdown
-				popoverProps={ POPOVER_PROPS }
-				className="block-editor-block-switcher"
-				contentClassName="block-editor-block-switcher__popover"
-				renderToggle={ ( { onToggle, isOpen } ) => {
-					const openOnArrowDown = ( event ) => {
-						if ( ! isOpen && event.keyCode === DOWN ) {
-							event.preventDefault();
-							event.stopPropagation();
-							onToggle();
-						}
-					};
-					const label =
-						1 === blocks.length
-							? __( 'Change block type or style' )
-							: sprintf(
-									/* translators: %s: number of blocks. */
-									_n(
-										'Change type of %d block',
-										'Change type of %d blocks',
-										blocks.length
-									),
-									blocks.length
-							  );
-
-					return (
-						<ToolbarGroup>
-							<ToolbarButton
-								className="block-editor-block-switcher__toggle"
-								onClick={ onToggle }
-								aria-haspopup="true"
-								aria-expanded={ isOpen }
-								title={ label }
-								onKeyDown={ openOnArrowDown }
-								showTooltip
-								icon={ <BlockIcon icon={ icon } showColors /> }
-							/>
-						</ToolbarGroup>
-					);
-				} }
-				renderContent={ ( { onClose } ) => (
-					<>
-						{ ( hasBlockStyles ||
-							possibleBlockTransformations.length !== 0 ) && (
-							<div className="block-editor-block-switcher__container">
-								{ hasBlockStyles && (
-									<MenuGroup>
-										<div className="block-editor-block-switcher__label">
-											{ __( 'Styles' ) }
-										</div>
-										<BlockStyles
-											clientId={ blocks[ 0 ].clientId }
-											onSwitch={ onClose }
-											onHoverClassName={
-												this.onHoverClassName
-											}
-										/>
-									</MenuGroup>
-								) }
-								{ possibleBlockTransformations.length !== 0 && (
-									<MenuGroup>
-										<div className="block-editor-block-switcher__label">
-											{ __( 'Transform to' ) }
-										</div>
-										<BlockTypesList
-											items={ possibleBlockTransformations.map(
-												( destinationBlockType ) => ( {
-													id:
-														destinationBlockType.name,
-													icon:
-														destinationBlockType.icon,
-													title:
-														destinationBlockType.title,
-												} )
-											) }
-											onSelect={ ( item ) => {
-												onTransform( blocks, item.id );
-												onClose();
-											} }
-										/>
-									</MenuGroup>
-								) }
+		const PreviewBlockPopover = (
```

Comment:
I suggest turning this into a function component and placing it somewhere in this module. That way, we avoid the unnecessary work when we evaluate this expression and don't actually render (`{ hoveredClassName !== null && PreviewBlockPopover }`).